### PR TITLE
RELATED: BB-1711 Measure value filters work with percentage measures

### PR DESCRIPTION
**Pull request by qduval**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5500/ : SUCCESS in 7m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/994/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5501/ : SUCCESS in 10m 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/995/ : SUCCESS in 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5529/ : SUCCESS in 11m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1037/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5531/ : SUCCESS in 4m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1040/ : SUCCESS in 4s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5559/ : SUCCESS in 11m 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1075/ : SUCCESS in 5s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2860/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1076/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2283/ : UNSTABLE in 16m 19s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5560/ : SUCCESS in 7m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1077/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2861/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1078/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2284/ : UNSTABLE in 16m 12s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5561/ : SUCCESS in 6m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1079/ : SUCCESS in 9s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1080/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2285/ : FAILURE in 1h 00m 19s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1081/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2286/ : FAILURE in 1h 00m 16s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5563/ : SUCCESS in 6m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1082/ : SUCCESS in 9s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2865/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1083/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2287/ : UNSTABLE in 15m 52s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5564/ : SUCCESS in 4m 36s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1084/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2866/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1085/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2288/ : UNSTABLE in 13m 04s

**Comment by qduval** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2867/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1086/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2289/ : SUCCESS in 12m 58s

File: src/components/filters/MeasureValueFilter/DropdownBody.tsx [hunk 1]
```diff
@@ -36,21 +39,28 @@ class DropdownBodyWrapped extends React.PureComponent<IDropdownBodyProps, IDropd
...
     public render() {
-        const { onCancel, intl } = this.props;
+        const { onCancel, warningMessage, intl } = this.props;
         const { operator } = this.state;
 
         return (
             <div className="gd-mvf-dropdown-body gd-dialog gd-dropdown overlay s-mvf-dropdown-body">
                 <div className="gd-mvf-dropdown-content">
+                    <div className="gd-mvf-dropdown-section">
+                        {warningMessage && (
```
Comment (xMort) [inline src/components/filters/MeasureValueFilter/DropdownBody.tsx]: Shouldn't condition include `gd-mvf-dropdown-section` as well? This way some unwanted paddings can be rendered.

File: examples/src/components/MeasureValueFilterDropdownPercentageExample.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        this.setState({ filters: [filter], displayDropdown: false });
+    };
+
+    onCancel = () => {
+        this.toggleButtonRef = null;
+        this.setState({ displayDropdown: false });
+    };
+
+    toggleDropdown = e => {
+        this.toggleButtonRef = !this.state.displayDropdown ? e.currentTarget : null;
```
Comment (yanlipnican) [inline examples/src/components/MeasureValueFilterDropdownPercentageExample.jsx]: I find this kind of hacky. Do it react way.
```
class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.myRef = React.createRef();
  }
  render() {
    return <div ref={this.myRef} />;
  }
}
```

File: examples/src/components/MeasureValueFilterDropdownRatioExample.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        this.setState({ filters: [filter], displayDropdown: false });
+    };
+
+    onCancel = () => {
+        this.toggleButtonRef = null;
+        this.setState({ displayDropdown: false });
+    };
+
+    toggleDropdown = e => {
+        this.toggleButtonRef = !this.state.displayDropdown ? e.currentTarget : null;
```
Comment (yanlipnican) [inline examples/src/components/MeasureValueFilterDropdownRatioExample.jsx]: Same as before

File: src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx [hunk 4]
```diff
@@ -69,6 +69,32 @@ describe("Measure value filter", () => {
         ).toEqual(true);
     });
 
+    it("should render an input suffix for each input field to display percentage sign if the measure is native percent", () => {
```
Comment (yanlipnican) [inline src/components/filters/MeasureValueFilter/tests/MeasureValueFilter.spec.tsx]: Split this to 2 tests for each operator

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5648/ : SUCCESS in 15m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1188/ : SUCCESS in 10s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5656/ : SUCCESS in 14m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1195/ : SUCCESS in 17s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2893/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1196/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2317/ : UNSTABLE in 19m 32s

**Comment by qduval** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5657/ : SUCCESS in 10m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1197/ : SUCCESS in 8s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2894/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1198/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2318/ : SUCCESS in 15m 50s

**Comment by yanlipnican** [comment]
LGTM

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2896/ : UNSTABLE in 17m 43s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5667/ : SUCCESS in 13m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1360/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1880/ : SUCCESS in 21m 09s

**Comment by qduval** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2898/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1209/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2321/ : UNSTABLE in 13m 47s

**Comment by qduval** [comment]
extended test - storybook

**Comment by yanlipnican** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2900/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/1211/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2322/ : SUCCESS in 15m 51s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2901/ : SUCCESS in 15m 45s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/5671/ : SUCCESS in 5m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1363/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1883/ : SUCCESS in 22m 14s

**Pull request merged**